core/txflow/src/txflow_task/beacon_witness_selector.rs

```diff
@@ -0,0 +1,59 @@
+use primitives::traits::WitnessSelector;
+use primitives::types::UID;
+use rand::thread_rng;
+use rand::prelude::IteratorRandom;
+use std::collections::HashSet;
+
+/// Witness selector used by the beacon chain where each new block index has a new set of witnesses.
+pub struct BeaconWitnessSelector {
+    witnesses: HashSet<UID>,
+    witnesses_ordered: Vec<UID>,
+    /// Witnesses without `owner_uid`.
+    other_witnesses: Vec<UID>,
+}
+
+impl BeaconWitnessSelector {
+    pub fn new(witnesses: HashSet<UID>, owner_uid: UID) -> Self {
+        let mut witnesses_ordered : Vec<_> = (&witnesses).into_iter().cloned().collect();
+        witnesses_ordered.sort();
+        let other_witnesses : Vec<_> = (&witnesses).into_iter().filter_map(|w|
+                if w == &owner_uid {
+                    None } else {
```

Comment:
Some strange style here. I would put the curly bracket after None on a new line

Reply:
That's how ternary operator is sometimes written in Rust. It is not an official standard, but I saw people doing it.